### PR TITLE
Feature/fetching restaurants

### PR DESCRIPTION
**Pull request by Cristianjs93**
### Description
In this pull request, we have implemented the functionality to make a GET request to the app API to retrieve a list of restaurants. The retrieved data is then used to render restaurant cards with relevant information on the frontend. Additionally, we have introduced filter functionality to refine the displayed restaurant list based on popularity, trendiness, and creation date.

### Changes Made

- Developed a new API integration function that performs a GET request to fetch a list of restaurants from the backend.
- Introducing changes on components that uses the retrieved restaurant data to dynamically render restaurant cards with relevant information.
- Created a filter function that allows users to sort and filter the restaurant list based on different criteria.

### Checklist

- [x] Implemented a function to fetch restaurant data from the backend API.
- [x] Designed a component to render restaurant cards with fetched information.
- [x] Developed a filter function to sort and display the restaurant list based on different criteria.
- [x] Ensured that the filter function works seamlessly with the rendered restaurant cards.

### CODE REVIEW

File: src/components/RestaurantCarousel/index.jsx [hunk 1]
```diff
@@ -1,38 +1,51 @@
...
+import { RestaurantsContext } from '../../store/RestaurantsContext';
+import './index.scss';
 
 const RestaurantsCarousel = () => {
+  const [loading, setLoading] = useState(true);
+
+  const { restaurants } = useContext(RestaurantsContext);
+
+  useEffect(() => {
+    (restaurants.length > 0) & setLoading(false);
```
Comment (ht1204) [inline src/components/RestaurantCarousel/index.jsx]: Given falsy value, replace by: `(restaurants.length) & setLoading(false);`

File: src/components/RestaurantsList/index.jsx [hunk 2]
```diff
@@ -1,76 +1,91 @@
...
+import { RestaurantsContext } from '../../store/RestaurantsContext';
+import './index.scss';
 
 const RestaurantsList = () => {
+  const [loading, setLoading] = useState(true);
+
+  const { restaurants } = useContext(RestaurantsContext);
+
+  useEffect(() => {
+    (restaurants.length > 0) & setLoading(false);
```
Comment (ht1204) [inline src/components/RestaurantsList/index.jsx]: same here,

File: src/store/RestaurantsContext.jsx [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+
+      setRestaurants(popularRestaurants);
+    } else if (filter == 'trend') {
+      const trendingRestaurants = data.filter(
+        (restaurant) => restaurant.trending
+      );
+
+      setRestaurants(trendingRestaurants);
+    } else if (filter == 'latest') {
+      const compareTime = new Date('2023-08-23T14:15:00');
```
Comment (ht1204) [inline src/store/RestaurantsContext.jsx:31]: Guess this is for current date comparison testing.

File: src/components/RestaurantCarousel/index.jsx (hunk 1 above)
Comment (Cristianjs93) [reply src/components/RestaurantCarousel/index.jsx]: Resolved

File: src/components/RestaurantsList/index.jsx (hunk 2 above)
Comment (Cristianjs93) [reply src/components/RestaurantsList/index.jsx]: Resolved

File: src/store/RestaurantsContext.jsx (hunk 3 above)
Comment (Cristianjs93) [reply src/store/RestaurantsContext.jsx:31]: You're right.

**Review by ht1204 — APPROVED** [review]

**Pull request merged**